File: app/src/main/java/com/zhan/transparenttoolbar/MainActivity.java
package com.zhan.transparenttoolbar;

import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import zhan.transparent.OnTransparentListener;
import zhan.transparent.widget.TransparentFrameLayout;

public class MainActivity extends AppCompatActivity {

  private static final String TAG = MainActivity.class.getSimpleName();

  private TransparentFrameLayout mTransparentFrameLayout;
  private RecyclerView mRecyclerView;
  private TextView numTv;

  private ImageView picIv;
  private TextView titleTv;
  private View line;

  private MyAdapter mMyAdapter;

  private int mDy;

  @Override protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);
    initView();
    initData();
    initListener();
  }

  private void initData() {
    LinearLayoutManager manager =
        new LinearLayoutManager(this, RecyclerView.VERTICAL, false);
    mRecyclerView.setLayoutManager(manager);

    mMyAdapter = new MyAdapter();
    mRecyclerView.setAdapter(mMyAdapter);

    mTransparentFrameLayout.setColorToBackGround(getResources().getColor(R.color.colorPrimary));
    mTransparentFrameLayout.setMaxOffset(getResources().getDimension(R.dimen.offset));
  }

  private void initListener() {
    mRecyclerView.addOnScrollListener(new RecyclerView.OnScrollListener() {
      @Override public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
        super.onScrolled(recyclerView, dx, dy);
        mDy += dy;
        mTransparentFrameLayout.updateTop(mDy);
      }
    });

    mTransparentFrameLayout.addOnScrollStateListener(new OnTransparentListener() {
      @Override public void onTransparentStart(float fraction) {  //scroll to top

      }

      @Override public void onTransparentEnd(float fraction) {  //scroll to max offset
        if (titleTv.getVisibility() == View.GONE) {
          titleTv.setVisibility(View.VISIBLE);
          picIv.setVisibility(View.VISIBLE);
          line.setVisibility(View.VISIBLE);
        }
      }

      @Override public void onTransparentUpdateFraction(float fraction) {  //scrolling...
        numTv.setText(fraction * 100 + "%");

        if (titleTv.getVisibility() == View.VISIBLE && fraction < 1) {
          titleTv.setVisibility(View.GONE);
          picIv.setVisibility(View.GONE);
          line.setVisibility(View.GONE);
        }
      }
    });
  }

  private void initView() {
    mTransparentFrameLayout = (TransparentFrameLayout) findViewById(R.id.tool_bar);
    mRecyclerView = (RecyclerView) findViewById(R.id.recycler_view);
    numTv = (TextView) findViewById(R.id.num_tv);
    titleTv = (TextView) findViewById(R.id.title_tv);
    picIv = (ImageView) findViewById(R.id.pic_iv);
    line = findViewById(R.id.line);
  }
}

File: app/src/main/java/com/zhan/transparenttoolbar/MainActivity.kt
package com.zhan.transparenttoolbar

import android.annotation.SuppressLint
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.activity_main.*
import zhan.transparent.OnTransparentListener

class MainActivity : AppCompatActivity() {

    private var mDy: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initData()
        initListener()
    }

    private fun initData() {
        val manager = LinearLayoutManager(this, RecyclerView.VERTICAL, false)
        recycler_view.layoutManager = manager

        val mMyAdapter = MyAdapter()
        recycler_view.adapter = mMyAdapter

        tool_bar.setColorToBackGround(ContextCompat.getColor(this, R.color.colorPrimary))
        tool_bar.setMaxOffset(resources.getDimension(R.dimen.offset))
    }

    private fun initListener() {
        recycler_view.addOnScrollListener(object : RecyclerView.OnScrollListener() {
            override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
                super.onScrolled(recyclerView, dx, dy)
                mDy += dy
                tool_bar.updateTop(mDy.toFloat())
            }
        })

        tool_bar.addOnScrollStateListener(object : OnTransparentListener {
            override fun onTransparentStart(fraction: Float) {  //scroll to top

            }

            override fun onTransparentEnd(fraction: Float) {  //scroll to max offset
                if (title_tv.visibility == View.GONE) {
                    title_tv.visibility = View.VISIBLE
                    pic_iv.visibility = View.VISIBLE
                    line.visibility = View.VISIBLE
                }
            }

            @SuppressLint("SetTextI18n")
            override fun onTransparentUpdateFraction(fraction: Float) {  //scrolling...
                num_tv.text = (fraction * 100).toInt().toString() + "%"
                if (title_tv.visibility == View.VISIBLE && fraction < 1) {
                    title_tv.visibility = View.GONE
                    pic_iv.visibility = View.GONE
                    line.visibility = View.GONE
                }
            }
        })
    }
}
